File: leetcode/121. Best Time to Buy and Sell Stock/solution.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        i, j, profit = 0, 1, 0
        while j < len(prices):
            if prices[i] > prices[j]:
                i = j
            else:
                profit = max(profit, prices[j] - prices[i])
            j += 1
        return profit

n = int(input())
prices = list(map(int, input().split()))
s = Solution()
print(s.maxProfit(prices))

File: leetcode/121. Best Time to Buy and Sell Stock/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int i = 0, j = 1, profit = 0;
        while (j < prices.size())
        {
            if (prices[i] > prices[j])
                i = j;
            else
                profit = max(profit, prices[j] - prices[i]);
            j++;
        }
        return profit;
    }
};

int main()
{
    int n;
    cin >> n;
    vector<int> v(n, 0);
    for (int i = 0; i < n; i++)
    {
        cin >> v[i];
    }
    Solution S;
    cout << S.maxProfit(v) << endl;
    return 0;
}
